### PR TITLE
chore: Update azure-aci to use nodeutil instead of node-cli

### PR DESCRIPTION
**Pull request by cpuguy83**
Fix webhook certificate auth with nodeutil
    
1. Routes were not setup and causing panics
2. The CA was not being loaded into the webhook auth and as such only
     token auth was available.

### CODE REVIEW

**Comment by cpuguy83** [comment]
Big diff due to undoing the revert.

File: cmd/virtual-kubelet/main.go [hunk 1]
```diff
@@ -16,71 +16,302 @@ package main
...
+		f.Name = "klog." + f.Name
+		flags.AddGoFlag(f)
+	})
+
+	flags.StringVar(&nodeName, "nodename", nodeName, "kubernetes node name")
+	flags.StringVar(&cfgPath, "provider-config", cfgPath, "cloud provider configuration file")
+	flags.StringVar(&clusterDomain, "cluster-domain", clusterDomain, "kubernetes cluster-domain")
+	flag.DurationVar(&startupTimeout, "startup-timeout", startupTimeout, "How long to wait for the virtual-kubelet to start")
+	flags.BoolVar(&disableTaint, "disable-taint", disableTaint, "disable the node taint")
+	flag.StringVar(&operatingSystem, "os", operatingSystem, "Operating System (Linux/Windows)")
```
Comment (helayoty) [inline cmd/virtual-kubelet/main.go:268]:
```suggestion
	flags.StringVar(&operatingSystem, "os", operatingSystem, "Operating System (Linux/Windows)")
```

File: cmd/virtual-kubelet/main.go [hunk 2]
```diff
@@ -16,71 +16,302 @@ package main
...
+	klog.InitFlags(klogFlags)
+	klogFlags.VisitAll(func(f *flag.Flag) {
+		f.Name = "klog." + f.Name
+		flags.AddGoFlag(f)
+	})
+
+	flags.StringVar(&nodeName, "nodename", nodeName, "kubernetes node name")
+	flags.StringVar(&cfgPath, "provider-config", cfgPath, "cloud provider configuration file")
+	flags.StringVar(&clusterDomain, "cluster-domain", clusterDomain, "kubernetes cluster-domain")
+	flag.DurationVar(&startupTimeout, "startup-timeout", startupTimeout, "How long to wait for the virtual-kubelet to start")
```
Comment (helayoty) [inline cmd/virtual-kubelet/main.go:266]: Not used anymore and needs to be removed.

File: cmd/virtual-kubelet/main.go [hunk 3]
```diff
@@ -16,71 +16,302 @@ package main
...
+			}
+			if webhookAuthzAuthedCacheTTL > 0 {
+				cfg.AuthzConfig.AllowCacheTTL = webhookAuthzAuthedCacheTTL
+			}
+			if webhookAuthzUnauthedCacheTTL > 0 {
+				cfg.AuthzConfig.AllowCacheTTL = webhookAuthzUnauthedCacheTTL
+			}
+
+			if clientCACert != "" {
+				ca, err := dynamiccertificates.NewDynamicCAContentFromFile("client-ca", clientCACert)
```
Comment (helayoty) [inline cmd/virtual-kubelet/main.go:146]: The client certificate in Virtual Kueblet named `ca-cert-bundle` as shown [here](https://github.com/virtual-kubelet/virtual-kubelet/blob/7c9bd20eea6f4b37929fb765ca80d69262ca51c5/cmd/virtual-kubelet/internal/commands/root/root.go#L186). Any problem with having different file name?

**Pull request closed without merging**